extensions/who-is-visiting-this-content/app.R

```diff
@@ -112,30 +121,55 @@ server <- function(input, output, session) {
   all_visits_data <- reactive({
     usage_data() |>
       filter(content_guid == input$content_guid) |>
+
+      # Compute time diffs and filter out hits within the session
+      group_by(user_guid) |>
+      mutate(time_diff = seconds(timestamp - lag(timestamp, 1))) |>
+      replace_na(list(time_diff = seconds(Inf))) |>
+      filter(time_diff > input$visit_lag_cutoff) |>
+      ungroup() |>
+
+      # Join to usernames
       left_join(user_names(), by = "user_guid") |>
       replace_na(list(full_name = "[Anonymous]")) |>
       arrange(desc(timestamp)) |>
       select(timestamp, full_name, username)
-  }) |> bindCache(date_range()$from_date, date_range()$to_date, input$content_guid)
+  })
 
   aggregated_visits_data <- reactive({
-    usage_data() |>
+    unfiltered_hits <- usage_data() |>
+      filter(content_guid == input$content_guid) |>
+      group_by(user_guid) |>
+      summarize(n_hits = n())
```

Comment:
I think in a final version of this, we might not even wanna show the unfiltered "hits" variable, as that might be more confusing. I'm leaving it in as we develop this so that we can see the difference the filtering makes.